applications/RANSApplication/custom_strategies/generic_convergence_criteria.cpp

```diff
@@ -0,0 +1,86 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+
+/* Project includes */
+#include "includes/define.h"
+#include "spaces/ublas_space.h"
+#include "utilities/openmp_utils.h"
+
+// Include base h
+#include "generic_convergence_criteria.h"
+
+namespace Kratos
+{
+template <>
+void GenericConvergenceCriteria<UblasSpace<double, CompressedMatrix, Vector>, UblasSpace<double, Matrix, Vector>>::CalculateConvergenceCheckNorms(
+    double& rSolutionNorm,
+    double& rIncreaseNorm,
+    double& rDofSize,
+    ModelPart& rModelPart,
+    DofsArrayType& rDofSet,
+    const TSystemMatrixType& rA,
+    const TSystemVectorType& rDx,
+    const TSystemVectorType& rb)
+{
+    KRATOS_TRY
+
+    int number_of_dofs = rDofSet.size();
+
+    double solution_norm{0.0}, increase_norm{0.0};
+    int dof_num{0};
+
+    // Set a partition for OpenMP
+    PartitionVector dof_partition;
+    const int number_of_threads = OpenMPUtils::GetNumThreads();
+    OpenMPUtils::DivideInPartitions(number_of_dofs, number_of_threads, dof_partition);
+
+    // Loop over Dofs
+#pragma omp parallel reduction(+ : solution_norm, increase_norm, dof_num)
+    {
```

Comment:
Using the parallel utils will avoid including the `openmp_utils.h`.

Reply:
It is planned :)